pkg/kubelet/cm/container_manager.go

```diff
@@ -118,6 +118,12 @@ type ContainerManager interface {
 
 	// UpdateAllocatedDevices frees any Devices that are bound to terminated pods.
 	UpdateAllocatedDevices()
+
+	// GetAllocatableDevices returns information about all the devices known to the manager
+	GetAllocatableDevices() podresourcesapi.ResourceDeviceMap
+
+	// GetAllocatableCPUs returns the assignable (not allocated) CPUs
+	GetAllocatableCPUs() []int64
```

Comment:
It feels a bit strange to have the `GetAllocatableDevices()` call return something very specific to the `podresourcesapi`. I would have expected a more generic type that then get's translated into the `podresourcesapi.ResourceDeviceMap` at a higher layer.

Reply:
good point, let me fix

Reply:
We introduced `podresourcesapi.ResourceDeviceMap` to abstract the type and avoid the awkward `map[string]map[string]pluginapi.Device` and to possibly avoid using the `pluginapi` dependency in the upper layers.
Moving the `map[string]map[string]pluginapi.Device` inside `devicemanager` is similar and probably a bit better because indeed we don't have a specific dependency on `podresourcesapi`.
So, if we go this direction, it makes little to none sense to have an intermediate step translating to `ResourceDeviceMap`. We can do just one translation to the public API value in the `podresources` server. 
Hence I removed this commit and simplified the series.